mixer/pkg/il/runtime/externs.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runtime
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"istio.io/istio/mixer/pkg/il/interpreter"
+)
+
+// Externs contains the list of standard external functions used during evaluation.
+var Externs = map[string]interpreter.Extern{
+	"ip":              interpreter.ExternFromFn("ip", externIp),
+	"ip_equal":        interpreter.ExternFromFn("ip_equal", externIpEqual),
+	"timestamp":       interpreter.ExternFromFn("timestamp", externTimestamp),
```

Comment:
Will we support standard comparisons operators ((==, >, etc) on timestamps and ip addresses too? Or is that not supported by CEL?

Reply:
I am still investigating these. The official definition have very limited support for comparison/equality operators:
https://github.com/google/cel-spec/blob/master/doc/langdef.md

Having said that, I am unearthing new information through discussions. I'll most likely open separate github issues to track these once I have a reasonable plan.